### PR TITLE
TAC_3032 Support TransferConsent events

### PR DESCRIPTION
**Pull request by rasmuslager**
onTransferConsentRequested
TransferConsentSucceeded
TransferConsendCancelled

### CODE REVIEW

File: Sources/DreamsDelegate.swift [hunk 1]
```diff
@@ -41,4 +41,10 @@ public protocol DreamsDelegate: AnyObject {
      Use this delegate callback to close or dismiss the dreams viewcontroller.
      */
     func handleExitRequest()
+    
+    /**
+        Called when a transfer requires a consent.
+     */
+
+    func handleDreamsTransferConsentRequested(requestId: String, consentId: String, completion: @escaping (Result<TransferConsentSuccess, TransferConsentError>) -> Void)
```
Comment (loffek) [inline Sources/DreamsDelegate.swift:49]: why handle**Dreams**TransferConsentRequested ?

**Review by loffek — COMMENTED** [review]
Looks good!
But please remind me how we deal with Error / Failure / Cancel here.
In order to cancel the request, the delegate runs the completion with `.failure(TransferConsentError(...)`. And in the future, if we need some other error state, we do `.failure(TransferConsentError(<some different reject reasons here>)`. Correct?

**Comment by rasmuslager** [comment]
> Looks good! But please remind me how we deal with Error / Failure / Cancel here. In order to cancel the request, the delegate runs the completion with `.failure(TransferConsentError(...)`. And in the future, if we need some other error state, we do `.failure(TransferConsentError(<some different reject reasons here>)`. Correct?

Yep we could add a reason to the TransferConsentError type, we dont support it at all right now though

File: Sources/DreamsDelegate.swift (hunk 1 above)
Comment (rasmuslager) [reply Sources/DreamsDelegate.swift:49]: taking a look at the whole file all of the functions except the one you see in the preview are named with dreams:P
https://github.com/dreamstechnology/dreams-ios-sdk/blob/465bb9a2a5772154f090a5d91424c578abcfaafb/Sources/DreamsDelegate.swift

Comment (loffek) [reply Sources/DreamsDelegate.swift:49]: Ha! True. Yeah, let's keep it consistent then 👍

**Review by loffek — APPROVED** [review]

**Pull request merged**